recipes/tree-gen/all/patches/0001-generate-tree.patch

```diff
@@ -0,0 +1,31 @@
+diff --git a/cmake/generate_tree.cmake b/cmake/generate_tree.cmake
+index 1a5a1f8..bad2bca 100644
+--- a/cmake/generate_tree.cmake
++++ b/cmake/generate_tree.cmake
+@@ -9,10 +9,11 @@ function(generate_tree TREE HDR SRC)
+     file(MAKE_DIRECTORY "${SRC_DIR}")
+ 
+     # Add a command to do the generation.
++    find_program(TREE_GEN_EXECUTABLE tree-gen REQUIRED)
```

Comment:
Wouldn't it be better if I just change this in the GitHub `tree-gen` package itself? So that we don't have to apply any patches?

Reply:
Yes, please. It would be much better!

Reply:
Hi @uilianries , I have a `tree-gen` 1.0.3 version that requests a `TREE_GEN_EXECUTABLE` parameter in the generate functions. Then, in a client such as `test_pacakge`, we can do `find_program(TREE_GEN_EXECUTABLE tree-gen REQUIRED)`, as you do in the patch file, and pass that variable to the generate functions. This saves us from using patches.